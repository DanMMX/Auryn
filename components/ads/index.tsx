import React, { createContext, createRef, RefObject, Component } from 'react';
import { StyleSheet, View } from 'react-native';

import PauseAd from './pause-ad';
import { VideoContext } from './../videoPlayer/context';

const END_SQUEEZE_MS = 15 * 1000;

interface AdProviderProps {
  pauseAdCompositionName: string | undefined;
  onPauseAdClosed: () => void;
}

interface AdProviderState {}

const Context = createContext<AdProviderState>({});

class AdProvider extends Component<AdProviderProps, AdProviderState> {
  static defaultProps = {
    pauseAdCompositionName: undefined,
    related: [],
    onPauseAdClosed: () => {}
  };

  static contextType = VideoContext;

  private pausAdRef: RefObject<PauseAd> = createRef();

  constructor(props: AdProviderProps) {
    super(props);
  }

  shouldComponentUpdate() {
    return false;
  }

  render() {
    const { children, pauseAdCompositionName, onPauseAdClosed } = this.props;
    
    return (
      <Context.Provider value={this.state}>
        <View style={styles.container}>
          {pauseAdCompositionName !== undefined ?
            <PauseAd
              ref={this.pausAdRef}
              name={pauseAdCompositionName}
              forceShow={this.context.duration - this.context.currentTime < END_SQUEEZE_MS}
              onClose={onPauseAdClosed}
            /> :
            null
          }
        </View>
        {children}
      </Context.Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    ...StyleSheet.absoluteFillObject
  }
});

export default AdProvider;
