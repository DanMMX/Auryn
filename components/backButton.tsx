/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { ButtonRef, ButtonRefProps } from '@youi/react-native-youi';
import { Config } from '../config';

export const BackButton: React.FunctionComponent<Partial<ButtonRefProps>> = ({ focusable, ...otherProps }) => (
  <ButtonRef
    {...otherProps}
    name="Btn-Back"
    visible={!Config.hasHardwareBackButton}
    focusable={focusable && !Config.hasHardwareBackButton}
  />
);
