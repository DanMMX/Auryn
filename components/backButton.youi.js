import React, { PureComponent } from 'react';
import { ButtonRef } from '@youi/react-native-youi';

export default class BackButton extends PureComponent {
  constructor(props) {
    super(props);
  }

  render = () =>
    <ButtonRef
      {...this.props}
      name="Btn-Back"
      visible={!this.props.hasBackButton}
      focusable={this.props.focusable && !this.props.hasBackButton}
    />
}
