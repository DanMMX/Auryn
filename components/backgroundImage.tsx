/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { View, Image, StyleSheet, ImageURISource } from 'react-native';

interface BackgroundImageProps {
  source: ImageURISource;
}

export const BackgroundImage: React.FunctionComponent<BackgroundImageProps> = ({ source, children }) => (
  <View style={styles.container}>
    <Image style={styles.background} source={source} />
    {children}
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 3,
  },
  background: {
    width: 1920,
    height: 1080,
    resizeMode: 'repeat',
    position: 'absolute',
    borderColor: 'black',
    borderWidth: 5,
  },
});
