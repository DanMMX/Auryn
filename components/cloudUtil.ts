/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { NativeModules, findNodeHandle } from 'react-native';

const { Cloud } = NativeModules;

export const CloudUtil: { isRoku: boolean; updateScene: (component: React.RefObject<any>) => void } = {
  updateScene: component =>
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), true);
      Cloud.sendFocusMap();
    }, 300),
  isRoku: Cloud.isCloudServer,
};

