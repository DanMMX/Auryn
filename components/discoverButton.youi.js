import React from 'react';
import { Composition, TextRef, ButtonRef, ImageRef } from '@youi/react-native-youi';

export default DiscoverButton = props => (
  <Composition source={"Auryn_Container-Btn-Main-" + props.type}>
    <ButtonRef name={"Btn-Main-" + props.type}>
      <ImageRef name="Image-Dynamic" source={{ uri: props.type == "Half" ? "http://image.tmdb.org/t/p/w500" + props.data.backdrop_path : "http://image.tmdb.org/t/p/w1280" + props.data.backdrop_path }} />
      <TextRef name="Text-Details" text={props.data.overview}/>
      <TextRef name="Text-Title" text={props.data.title}/>
    </ButtonRef>
  </Composition>
);


//"http://image.tmdb.org/t/p/w500" + props.data.backdrop_path
