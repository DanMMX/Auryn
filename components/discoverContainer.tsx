/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { View } from 'react-native';
import { ListItem } from '.';
import { Asset } from '../adapters/asset';

interface DiscoverContainerProps {
  data: Asset[];
  onPress: () => void;
  onFocus: () => void;
  focusable: boolean;
  index: number;
}

// eslint-disable-next-line max-lines-per-function
export const DiscoverContainer: React.FunctionComponent<DiscoverContainerProps> = ({
  data,
  onPress,
  onFocus,
  focusable,
  index,
}) => {
  if (data.length !== 3) return null;

  const smallItems = (
    <View style={{ flexDirection: 'row' }}>
      <ListItem
        focusable={focusable}
        onPress={onPress}
        onFocus={onFocus}
        shouldChangeFocus={index % 2 !== 0}
        imageType={{ type: 'Backdrop', size: 'Small' }}
        data={data[0]}
      />
      <ListItem
        focusable={focusable}
        onPress={onPress}
        onFocus={onFocus}
        shouldChangeFocus={index % 2 !== 0}
        imageType={{ type: 'Backdrop', size: 'Small' }}
        data={data[1]}
      />
    </View>
  );
  const largeItem = (
    <ListItem
      focusable={focusable}
      onPress={onPress}
      onFocus={onFocus}
      shouldChangeFocus={index % 2 === 0}
      imageType={{ type: 'Backdrop', size: 'Large' }}
      data={data[2]}
    />
  );
  if (index % 2) {
    return (
      <View>
        {smallItems}
        {largeItem}
      </View>
    );
  }

  return (
    <View>
      {largeItem}
      {smallItems}
    </View>
  );
};
