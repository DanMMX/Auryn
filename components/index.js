/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export { BackButton } from './backButton';
import ListItem from './listitem.youi';
import List from './list.youi';
import Timeline from './timeline.youi';
import DiscoverContainer from './discoverContainer.youi';
import ToggleButton from './toggleButton.youi';
import ToggleGroup from './toggleGroup.youi';
import TvContainer from './tvContainer.youi';
import BackgroundImage from './backgroundImage.youi';

export {
  ListItem,
  List,
  Timeline,
  DiscoverContainer,
  ToggleButton,
  ToggleGroup,
  TvContainer,
  BackgroundImage,
};
