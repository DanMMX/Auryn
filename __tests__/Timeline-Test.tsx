/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { Timeline } from '../components';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Timeline loop={false} name="In"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
