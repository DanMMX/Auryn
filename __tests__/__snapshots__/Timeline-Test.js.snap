// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TimelineRef
  direction="forward"
  loop={false}
  name="In"
  onAborted={[Function]}
  onCompleted={[Function]}
  onCompletedForward={[Function]}
  onCompletedReverse={[Function]}
  onLoad={[Function]}
  onPaused={[Function]}
  onPausedForward={[Function]}
  onPausedReverse={[Function]}
  onPlay={[Function]}
  onPlayForward={[Function]}
  onPlayReverse={[Function]}
  onStarted={[Function]}
  onStartedForward={[Function]}
  onStartedReverse={[Function]}
  playOnLoad={false}
/>
`;
